File: CWM/Models/WorkOrderInsertUpdate.cs
using CWM.Core.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace CWM.Models
{
    public class WorkOrderInsertUpdate
    {
        public string OrderNumber { get; set; } = string.Empty;
        public decimal Total {  get; set; }
        public DateTime? StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public GarageBox GarageBox { get; set; }
        public Service ServicePerformed { get; set; }
        public string Concerne { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Sugestions { get; set; } = string.Empty;

        [Required]
        public int? VehicleId { get; set; }
        [Required]
        public int? AppointmentId { get; set; }
        [Required]
        public int? UserId { get; set; }
        [Required]
        public int? EmployeeId { get; set; }
    }
}

File: CWM.UI/cwm_desktop_mobile/lib/models/requests/work_order_insert_update.dart
import 'package:cwm_desktop_mobile/models/enums/garage_box.dart';
import 'package:cwm_desktop_mobile/models/enums/service.dart';
import 'package:json_annotation/json_annotation.dart';

part 'work_order_insert_update.g.dart';

@JsonSerializable()
class WorkOrderInsertUpdate {
  String orderNumber;
  double total;
  bool payment;
  DateTime startTime;
  DateTime endTime;
  GarageBox garageBox;
  Service servicePerformed;
  String concerne;
  String description;
  String sugestions;
  String? vehicleId;
  String? userId;
  String? appointmentId;
  String? employeeId;

  WorkOrderInsertUpdate(
      this.orderNumber,
      this.total,
      this.payment,
      this.startTime,
      this.endTime,
      this.garageBox,
      this.servicePerformed,
      this.concerne,
      this.description,
      this.sugestions,
      this.vehicleId,
      this.userId,
      this.appointmentId,
      this.employeeId);

  factory WorkOrderInsertUpdate.fromJson(Map<String, dynamic> json) =>
      _$WorkOrderInsertUpdateFromJson(json);

  Map<String, dynamic> toJson() => _$WorkOrderInsertUpdateToJson(this);
}
